### PR TITLE
feat: migrate frontend framework from mdbook to vocs 

### LINKED ISSUE #167: Improvement: Migrating to `vocs` static site generator
Migration to [wevm/vocs](https://github.com/wevm/vocs).
Current progress on branch [vocs](https://github.com/NethermindEth/StarknetByExample/tree/vocs)

### PR DESCRIPTION
**Pull request by kushagrasarathe**
**Issue(s):**  Closes #167 

### Description

Frontend migrated from mdbook to vocs

### CODE REVIEW

File: .tool-versions
Comment (julio4) [inline .tool-versions:1]: Don't delete this file

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -31,80 +28,34 @@ Please note we have a code of conduct, please follow it in all your interactions
 
 1. Clone this repository.
 
-2. Rust related packages:
-   - Install toolchain providing `cargo` using [rustup](https://rustup.rs/).
-   - Install [mdBook](https://rust-lang.github.io/mdBook/guide/installation.html) and the required extension with `cargo install mdbook  mdbook-i18n-helpers mdbook-last-changed `.
-
-3. Install `scarb` using [asdf](https://asdf-vm.com/) with `asdf install`. Alternatively, you can install `scarb` manually by following the instructions [here](https://docs.swmansion.com/scarb/).
-
```
Comment (julio4) [inline CONTRIBUTING.md:39]: We still need scarb

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -11,14 +11,11 @@ Please note we have a code of conduct, please follow it in all your interactions
 - [Contributing](#contributing)
   - [Table of Contents](#table-of-contents)
   - [Setup](#setup)
-  - [MdBook](#mdbook)
+  - [Working with Markdown Files](#working-with-markdown-files)
   - [Adding a new chapter](#adding-a-new-chapter)
   - [Adding a new Cairo program](#adding-a-new-cairo-program)
-    - [Verification script](#verification-script)
```
Comment (julio4) [inline CONTRIBUTING.md:17]: Don't remove the Verification script part

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -31,80 +28,34 @@ Please note we have a code of conduct, please follow it in all your interactions
...
+  ```
+  pnpm build
+  ```
+- If everything looks good, commit your changes and open a pull request with your modifications.
 
 ## Adding a new chapter
 
-To add a new chapter, create a new markdown file in the `src` directory. All the Markdown files **MUST** be edited in english. In order to add them to the book, you need to add a link to it in the `src/SUMMARY.md` file.
-
-Do not write directly Cairo program inside the markdown files. Instead, use code blocks that import the Cairo programs from the `listing` directory. These programs are bundled into scarb projects, which makes it easier to test and build all programs. See the next section for more details.
```
Comment (julio4) [inline CONTRIBUTING.md:55]: Keep the `Do not write directly Cairo program inside the markdown files` paragraph.

File: scripts/cairo_programs_verifier.sh
Comment (julio4) [inline scripts/cairo_programs_verifier.sh:1]: Do not delete `scripts` repository

File: docs/pages/index.mdx [hunk 4]
```diff
@@ -34,6 +39,7 @@ For more resources, check [Awesome Starknet](https://github.com/keep-starknet-st
 ## Versions
 
 The current version of this book use:
+
 ```
 cairo 2.6.3
 edition = '2023_11'
```
Comment (julio4) [inline docs/pages/index.mdx:45]: Forgot include of scarb version

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -31,80 +28,34 @@ Please note we have a code of conduct, please follow it in all your interactions
...
-Instead of directly writing Cairo programs in the markdown files, we use code blocks that import the Cairo programs from the `listing` directory.
-These programs are bundled into scarb packages, which makes it easier to test and build entire packages.
-
-To run the script locally, ensure that you are at the root of the repository, and run:
-
-`bash scripts/cairo_programs_verifier.sh`
-
-This will check that all the Cairo programs in the book compile successfully using `scarb build`, that every tests passes using `scarb test`, and that the `scarb fmt -c` command does not identify any formatting issues.
-
-You can also use `scarb fmt` to format all the Cairo programs.
```
Comment (julio4) [inline CONTRIBUTING.md:107]: Keep this

File: docs/pages/ch00/basics/constructor.md [hunk 6]
```diff
@@ -1,9 +1,24 @@
+---
+content:
+  horizontalPadding: 50px
+  width: 100%
+  verticalPadding: 30px
+---
+
```
Comment (julio4) [inline docs/pages/ch00/basics/constructor.md:7]: Is there a way to modify theme/page template to not have to add this styling header on each pages?

**Review by julio4 — CHANGES_REQUESTED** [review]
Great! Is there a reason why you moved the `listings` directory to `docs/snippets`?
Default `rootDir` is `./docs`, but it can be changed in vocs.config.ts. You should try to change it to`rootDir: '.'` and see if we can keep `listings` and `docs` separated.

That way we can import the repository either as a scarb project (`listings`) or as a vocs project (`docs`)

File: docs/pages/ch00/basics/constructor.md (hunk 6 above)
Comment (kushagrasarathe) [reply docs/pages/ch00/basics/constructor.md:7]: Looked into this, couldn't find anything specific in docs regarding this. Will look more and update you

**Review by julio4 — APPROVED** [review]

**Pull request merged**